### PR TITLE
improvement over PR/14

### PR DESCRIPTION
**Pull request by adon-at-work**
This is to improve https://github.com/yahoo/html-purify/pull/14:
- a user customizing whitelist should be able to really stop style attributes
- fix a vulnerable [test case](https://github.com/yahoo/html-purify/commit/b63bd05e11cb757cd2fdfb0c62d2486269160aa0#diff-13093af8b7bfab50b4de2535c8914aa7R58)
- converted attribute list to lower case, otherwise, purifier can be bypassed
- simplified the config logics, it's reasonable to skip copying things around in constructor
- hrefAttributes should be a hardcoded list (ultimately should rely on a single source of truth and a [tag-specific list](https://github.com/yahoo/secure-handlebars/blob/master/src/parser-utils.js#L82)) to judge whether url filter is needed

### CODE REVIEW

**Comment by adon-at-work** [comment]
@maditya , @neraliu , please review.
note: this PR is designed to be first absorbed by taglist-configurable, which will ultimately be (rebased and) merged into master.

**Comment by yahoocla** [comment]
CLA is valid!

File: src/tag-attr-list.js [hunk 1]
```diff
@@ -121,82 +119,82 @@ Whitelist.Tags = [
...
-    "autoPlay",
-    "cellPadding",
-    "cellSpacing",
-    "charSet",
+    "autocomplete",
+    // "autofocus",
+    "autoplay",
+    "cellpadding",
+    "cellspacing",
+    // "charset",
```
Comment (adon-at-work) [inline src/tag-attr-list.js:136]: I commented out autofocus and charset temporarily in order to pass their corresponding test cases. They were bypassed by the test cases before this list is lowercased.

Let's discuss the default whitelist tags and attributes. I imagine we'd have a much simpler list.
The current setting is vulnerable. For instance, it enables `<iframe srcdoc="">`, which is actually scriptable. Allowing `<iframe src="http://attacker.com">` is not ideal too.

**Comment by maditya** [comment]
Merging all these changes via cli into https://github.com/yahoo/html-purify/pull/14 and fixing pending items. Closing the PR

File: src/tag-attr-list.js (hunk 1 above)
Comment (maditya) [inline src/tag-attr-list.js:136]: >  Allowing `<iframe src="http://attacker.com">` is not ideal too.

We can handle this better by allowing user to specify a whitelist of domains. For eg: we should allow  `<iframe src="http://mydomain.com">`  but filter the value for `<iframe src="http://attacker.com">`

**Pull request closed without merging**